docs/installation/local.md

```diff
@@ -2,169 +2,183 @@
+### For Mac OS
 
-# For macOS users
+```sh
+brew install postgresql
 brew install python@3
 brew install libmagic
+brew install redis
 ```
 
-## Using pip and virtualenv
+**Note:** For Mac OS Sierra users, if you get an error that 'openssl/aes.h' could not be found when installing requirements.txt using pip, try the steps shown here - [OSX openssl header error](https://tutorials.technology/solved_errors/1-OSX-openssl_opensslv_h-file-not-found.html)
+
+### For Debian/Ubuntu
 
-Open a terminal and enter the following commands to setup a virtual environment
+The dependencies are listed in *deb-packages.txt* file. You can install them all with one command:
 
 ```sh
-sudo apt-get install python3.7
-virtualenv -p python3.7 venv
-. venv/bin/activate
+xargs -a deb-packages.txt sudo apt install
 ```
 
-Now to install the dependencies using pip, type
+In case you use Ubuntu 20.04+, where Python 3.7 is not provided in official repo, you can use [pyenv](https://github.com/pyenv/pyenv) to install Python 3.7 (Open Event Server is not compatible with Python 3.8+ yet).
 
-```sh
-pip3 install -r requirements.txt
-```
+## Create a Python virtual environment
 
-## Using pipenv
+Python virtual environment is a way to keep this project's collection of libraries not interfere with other projects or the system. There are many ways to create Python virtual environment (all have the same core), like:
 
-Using pipenv, you will not need to set up virtualenv. It will do it automatically for you
+- Use Python built-in tool:
 
-To setup a virtual environment and install the dependices, enter in a terminal
+  ```sh
+  python3 -m venv .venv
+  ```
 
-```sh
-pipenv --python 3.7.3 install
-```
+  Running this command will create a folder named *.venv* in your working copy. Please remember
+  not to commit it.
+
+
+- Use third-party tools:
+
+  ```sh
+  mkvirtualenv my-project-name -p /usr/bin/python3
+  ```
 
-Now to activate the virtual environment, type
+I (@hongquan) recommend [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/) because:
+
+- It gathers all virtual environments to one place. That let you know which virtual environments you created.
+- It provides auto-complete command to enter a virtual env (`workon my-env`), list and delete them.
+- It let you enter a virtual env no matter where you are standing.
+
+Some tools, like Pipenv, puts virtual environment into your project code folder. That approach has some disadvantage:
+
+- When you do somethings that involve scanning source code (like running test cases, search for file content), you accidentally scan the Python packages in the virtual environment.
+- You don't know how many virtual environments you created.
+- You have to know the path of an virtual environment in order to enter (if you are not standing in project folder).
+
+After creating virtual environment, you should upgrade `pip`, `wheel` libraries inside it.
 
 ```sh
-pipenv shell
+pip install -U pip wheel setuptools
```

Comment:
This is already installed in the virtualenv by default

Reply:
Of course it is installed in the virtualenv by default. But please notice the swtich `-U`. It means "upgrade". I see many cases that the existing "pip", "setuptools" aren't new enough and make later `pip install` fail.